unreleased/value_parsing.yaml

```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```

Comment:
Can you enlighten me on why this is a breaking change? Isn't this just adding a new feature?

Reply:
Got it, we might need to do this change in two steps then, along with: a) the description you just gave should be part of the changelog, b) the migration path: what users should do if they are badly affected by this change?

Reply:
If user attention is required in order to not break a current setup, then it does need to be done in two steps, as described in the contribution guidelines, on the part about backward compatibility.

Reply:
Sure so lets make a decision here.  I personally am a fan of strong typing so I would vote to keep the current impl, however I understand this is a matter of opinion and not driven by requirements.  So @jpkrohling and @Aneurysm9 please throw your vote into the hat.  If you both wanna go the string way I can do it in about 10 seconds its only two lines of code from the original

Reply:
@jpkrohling??????? if nobody says anything I am going with this impl.